### ISSUE TITLE
oauth parameters is not passed properly

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Dragon-Born/google-api-python-client#1]
```
I use google-api-python-client in my application to access google latitude data.
First, I must get the authorization token, I do this by the following lines:

flow = FlowThreeLegged(latitude_discovery,
                       consumer_key='my_key',
                       consumer_secret='my_secret',
                       user_agent='my_user_agent',
                       domain='my_domain',
                       scope='https://www.googleapis.com/auth/latitude',
                       xoauth_displayname='Google API Client Example App',
                       location='all',
                       granularity='best'
                    )

I pass two parameters location and granularity to the FlowThreeLegged class, 
and it must send to the oauth url, but not.

I think the problem happens in oauth.py line 56-60:

  keys = discovery[name]['parameters'].keys()
  query = {}
  for key in keys:
    if key in params:
      query[key] = params[key]

keys did not contain location or granularity, neither the url.

It seems that the file "/contrib/latitude/future.json" is out-of-date.
```

Original issue reported on code.google.com by `win...@gmail.com` on 18 Oct 2010 at 8:24

**Comment by GoogleCodeExporter** [comment]
```
Thank you for reporting this. We will investigate and get back to you.
```

Original comment by `a...@google.com` on 22 Oct 2010 at 10:26
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision c8b52f84bb.
```

Original comment by `jcgregorio@google.com` on 25 Oct 2010 at 7:28
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report, and yes that was indeed where the problem was. I've just 
checked in a fix.
```

Original comment by `jcgregorio@google.com` on 25 Oct 2010 at 7:28
